sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifact.java

```diff
@@ -34,23 +53,68 @@
  */
 @ServiceClient(builder = ContainerRegistryClientBuilder.class)
 public final class RegistryArtifact {
-    private final RegistryArtifactAsync asyncClient;
+    private final ClientLogger logger = new ClientLogger(RegistryArtifact.class);
+    private final ContainerRegistriesImpl serviceClient;
+    private final String repositoryName;
+    private final String fullyQualifiedReference;
+    private final String endpoint;
+    private final String tagOrDigest;
+    private String digest;
 
     /**
      * Creates a RegistryArtifact type that sends requests to the given repository in the container registry service at {@code endpoint}.
      * Each service call goes through the {@code pipeline}.
-     * @param asyncClient The async client for the given repository.
+     * @param repositoryName The name of the repository on which the service operations are performed.
+     * @param tagOrDigest The tag or digest associated with the given artifact.
+     * @param endpoint The URL string for the Azure Container Registry service.
+     * @param httpPipeline HttpPipeline that the HTTP requests and responses flow through.
+     * @param version {@link ContainerRegistryServiceVersion} of the service to be used when making requests.
      */
-    RegistryArtifact(RegistryArtifactAsync asyncClient) {
-        this.asyncClient = asyncClient;
+    RegistryArtifact(String repositoryName, String tagOrDigest, HttpPipeline httpPipeline, String endpoint, String version) {
```

Comment:
Please consider https://github.com/samvaity/azure-sdk-for-java/pull/4 that allows to reduce duplication